packages/core/strapi/lib/services/entity-service/types/index.d.ts

```diff
@@ -1,14 +1,36 @@
-import type { Attribute, Common, EntityService } from '@strapi/strapi';
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+import type { Result, PaginatedResult } from './result';
+
+import type * as Params from './params';
 
 // TODO: Move params to @strapi/utils (related to convert-query-params)
 export * as Params from './params';
 
+export * from './result';
 export * from './plugin';
 
+type WrapAction = Omit<keyof EntityService, 'wrapParams' | 'wrapResult' | 'emitEvent'>;
+
 export interface EntityService {
+  wrapParams<TContentTypeUID extends Common.UID.ContentType, TParams extends object>(
+    params?: TParams,
+    options?: { uid: TContentTypeUID; action: WrapAction }
+  ): unknown;
+
+  wrapResult<TContentTypeUID extends Common.UID.ContentType>(
+    result: any,
+    options?: { uid: TContentTypeUID; action: WrapAction }
+  ): any;
+
+  emitEvent<TContentTypeUID extends Common.UID.ContentType>(
+    uid: TContentTypeUID,
+    event: Event,
+    entity: Attribute.GetValues<TContentTypeUID> // double check
```

Comment:
We need to delete this 😅

Reply:
```suggestion
    entity: Partial<Attribute.GetValues<TContentTypeUID>>
```

I double-checked, and it brings up a problem we may have elsewhere (although I know we are addressing it for return values in a later PR, so maybe not)

The entity passed in might be a partial entity without all the required fields, so they won't exist on the object.

Reply:
Hence the other task about the "smart" return type (based on params)
We could either keep those Attribute.GetValues while we wait for ID & required/optional props or switch to an any with a comment.

What do you think?

Reply:
We need to PartialDeep instead then, otherwise relations & components will still have required fields